userapi/internal/device_list_update.go

```diff
@@ -405,24 +418,40 @@ func (u *DeviceListUpdater) worker(ch chan spec.ServerName, workerID int) {
 	go func() {
 		var serversToRetry []spec.ServerName
 		for {
+			deviceListUpdaterServersRetrying.With(prometheus.Labels{"worker_id": strconv.Itoa(workerID)}).Set(float64(len(retries)))
+			time.Sleep(time.Second * 2)
+
+			// The channel is at capacity, don't try to send more work
+			if len(ch) == cap(ch) {
+				continue
+			}
 			serversToRetry = serversToRetry[:0] // reuse memory
```

Comment:
This is super unclear that it's nuking the `serversToRetry` slice, and why. Can we add more comments here please?